sharktank/sharktank/layers/mixture_of_experts_block.py

```diff
@@ -33,6 +33,7 @@ def __init__(
         rms_epsilon: float,
         moe_activation=torch.nn.functional.silu,
         *,
+        experts_ffn_moe_block: PreGatherFFNMOE | DenseFFNMOE | str = "DenseFFNMOE",
```

Comment:
We should either stick to dependency injection OR enum configuration. Supporting both is a recipe for difficult maintenance.